src/pages-and-resources/discussions/app-list/AppListNextButton.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { useCallback, useContext } from 'react';
+import { history } from '@edx/frontend-platform';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Button } from '@edx/paragon';
+import { useSelector } from 'react-redux';
+
+import { DiscussionsContext } from '../DiscussionsProvider';
+
+import messages from './messages';
+
+function AppListNextButton({ intl }) {
+  const { selectedAppId } = useSelector(state => state.discussions.appList);
+  const { path: discussionsPath } = useContext(DiscussionsContext);
+
+  const handleStartConfig = useCallback(() => {
+    history.push(`${discussionsPath}/configure/${selectedAppId}`);
+  }, [discussionsPath, selectedAppId]);
+
+  return (
+    <Button variant="primary" onClick={handleStartConfig}>
+      {intl.formatMessage(messages.nextButton)}
+    </Button>
+  );
+}
+
+AppListNextButton.propTypes = {
+  intl: intlShape.isRequired,
```

Comment:
I can see `intl` is required here as a prop when calling `AppListNextButton`. I am not sure where is this being used and intl prop is being passed. If you can help clarify the flow that would be great.

Reply:
Yeah, this is injected via a "higher order function", which is a weird pattern that's common in many React components.  On line 30 below you can see `injectIntl` wraps the export of AppListNextButton.  It's responsible for providing this `intl` prop to the component so that the parent component doesn't have to do it.  You'll see this in pretty much every single component that uses any i18n messages in it.